tsl/src/continuous_aggs/create.c

```diff
@@ -720,16 +720,82 @@ cagg_validate_query(Query *query)
 				 errmsg("invalid continuous aggregate query"),
 				 errhint("Use a SELECT query in the continuous aggregate view.")));
 	}
-	if (query->hasWindowFuncs || query->hasSubLinks || query->hasDistinctOn ||
-		query->hasRecursive || query->hasModifyingCTE || query->hasForUpdate ||
-		query->hasRowSecurity || query->hasTargetSRFs || query->cteList || query->groupingSets ||
-		query->distinctClause || query->setOperations || query->limitOffset || query->limitCount ||
-		query->sortClause)
+
+	if (query->hasWindowFuncs)
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("invalid continuous aggregate view")));
+				 errmsg("invalid continuous aggregate view"),
```

Comment:
Only the details vary here. It is probably better to have a function that accept the `query` and produces the error details separately.